### PR TITLE
Set up GraphQL

### LINKED ISSUE #5: Set up GraphQL

### PR DESCRIPTION
**Pull request by tomekpiotrowski**
Sets up GraphQL Playground at localhost:8080. Allows simple querying of ledgers.

Fixes #5

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -42,44 +39,6 @@ struct Args {
...
-    let config = Log4rsConfig::builder()
-        .appender(Appender::builder().build("stdout", Box::new(stdout)))
-        .build(Root::builder().appender("stdout").build(LevelFilter::Info))
-        .unwrap();
-
-    log4rs::init_config(config).unwrap();
-}
-
-fn setup_configuration(args: Args) -> Configuration {
-    let mut config_builder = Config::builder()
```
Comment (geofmureithi) [inline src/main.rs:57]: We might want to use [Figment](https://docs.rs/figment/latest/figment/) here, which can also handle args without the overrides. Not critical to implement though.

File: quasar_entities/src/ledger.rs [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+
 use stellar_node_entities::ledgerheaders;
 use stellar_xdr::{Error, LedgerHeader, ReadXdr};
 
-#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, async_graphql::SimpleObject)]
 #[sea_orm(table_name = "ledgers")]
+#[graphql(complex)]
+#[graphql(name = "Ledgers")]
 pub struct Model {
```
Comment (geofmureithi) [inline quasar_entities/src/ledger.rs:11]: Could we rename entities from `Model` to a representative name eg Ledger?

File: src/server.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        "/",
+        get(graphql_playground).post_service(GraphQL::new(schema)),
+    );
+
+    let socket_addr = format!("{}:{}", api.host, api.port)
+        .parse()
+        .expect("Failed to parse socket address");
+
+    tokio::spawn(async move {
+        axum::Server::bind(&socket_addr)
```
Comment (geofmureithi) [inline src/server.rs:32]: I think you should be able to pass in a tuple
```suggestion
        axum::Server::bind(&(api.host, api.port))
```

**Review by geofmureithi — APPROVED** [review]
LGTM.
Left some thoughts on logging and config. Let me know what you think.

File: src/main.rs (hunk 1 above)
Comment (tomekpiotrowski) [reply src/main.rs:57]: Wasn't familiar with it. I'll check it out.

File: src/main.rs [hunk 4]
```diff
@@ -42,44 +39,6 @@ struct Args {
     stellar_node_database_url: Option<String>,
 }
 
-fn setup_logger() {
-    let stdout = ConsoleAppender::builder().build();
-
-    let config = Log4rsConfig::builder()
```
[the remark this answers is not in the document]
Comment (tomekpiotrowski) [reply src/main.rs:48]: It seemed to be the most feature-rich of all the backends. Am I mistaken?

Tracing might be a good addition at some point.

File: quasar_entities/src/ledger.rs (hunk 2 above)
Comment (tomekpiotrowski) [reply quasar_entities/src/ledger.rs:11]: This seems to be the naming convention in SeaORM. The original of this file was actually generated by SeaORM CLI tooling. There's also `ActiveModel` and `Entity` that are generated by the macros. For now I'm not familiar enough with SeaORM to change the default naming conventions.

File: src/server.rs (hunk 3 above)
Comment (tomekpiotrowski) [reply src/server.rs:32]: I don't think `bind` accepts those arguments. Did you happen to mistaken `axum` for `actix-web`?

Comment (geofmureithi) [reply src/server.rs:32]: I think I thought they had the `Into<SocketAddr>` but seems we need to add an into.
```diff
diff --git a/src/configuration.rs b/src/configuration.rs
index fb9a589..e86219f 100644
--- a/src/configuration.rs
+++ b/src/configuration.rs
@@ -1,3 +1,5 @@
+use std::net::Ipv4Addr;
+
 use config::{Config, Environment, File};
 use serde::Deserialize;
 
@@ -8,16 +10,16 @@ pub struct Ingestion {
     pub polling_interval: u64,
 }
 
-#[derive(Debug, Default, Deserialize, PartialEq, Eq)]
+#[derive(Debug, Deserialize, PartialEq, Eq)]
 pub struct Api {
-    pub host: String,
+    pub host: Ipv4Addr,
     pub port: u16,
 
     pub depth_limit: usize,
     pub complexity_limit: usize,
 }
 
-#[derive(Debug, Default, Deserialize, PartialEq, Eq)]
+#[derive(Debug, Deserialize, PartialEq, Eq)]
 pub struct Configuration {
     pub quasar_database_url: Option<String>,
     pub stellar_node_database_url: Option<String>,
diff --git a/src/server.rs b/src/server.rs
index 31d6c91..758195b 100644
--- a/src/server.rs
+++ b/src/server.rs
@@ -24,9 +24,7 @@ pub async fn serve(api: &Api, database: DatabaseConnection) {
         get(graphql_playground).post_service(GraphQL::new(schema)),
     );
 
-    let socket_addr = format!("{}:{}", api.host, api.port)
-        .parse()
-        .expect("Failed to parse socket address");
+    let socket_addr = (api.host, api.port).into();
 
     tokio::spawn(async move {
         axum::Server::bind(&socket_addr)
```
This works. WDYT?

File: src/main.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (geofmureithi) [reply src/main.rs:48]: Axum and async-graphql have inbuilt support for tracing, we can add this later though.

File: src/server.rs (hunk 3 above)
Comment (tomekpiotrowski) [reply src/server.rs:30]: I've applied your suggestion

**Review by geofmureithi — APPROVED** [review]
LGTM

**Pull request merged**